internal/pkg/cli/project_init.go

```diff
@@ -79,25 +93,24 @@ func (o *initProjectOpts) Validate() error {
 
 // Ask prompts the user for any required arguments that they didn't provide.
 func (o *initProjectOpts) Ask() error {
-	// If there's a local project, we'll use that over anything else.
+	// When there's a local project
 	summary, err := o.ws.Summary()
 	if err == nil {
-		msg := fmt.Sprintf(
-			"Looks like you are using a workspace that's registered to project %s.\nWe'll use that as your project.",
-			color.HighlightResource(summary.ProjectName))
-		if o.ProjectName != "" && o.ProjectName != summary.ProjectName {
-			msg = fmt.Sprintf(
-				"Looks like you are using a workspace that's registered to project %s.\nWe'll use that as your project instead of %s.",
-				color.HighlightResource(summary.ProjectName),
-				color.HighlightUserInput(o.ProjectName))
+		if o.ProjectName == "" {
+			log.Infoln(fmt.Sprintf(
+				"Looks like you are using a workspace that's registered to project %s.\nWe'll use that as your project.",
+				color.HighlightResource(summary.ProjectName)))
+			o.ProjectName = summary.ProjectName
+			return nil
+		}
+		if o.ProjectName != summary.ProjectName {
+			// Error out if project name specified by flag is different from the one in workspace.
+			return fmt.Errorf("project init cancelled - name conflict between %s and local project name %s", o.ProjectName, summary.ProjectName)
```

Comment:
maybe: 
```go
log.Errorf(`Workspace is already registered with project %s instead of %s.
  If you'd like to delete the project locally, you can remove the %s directory.
  If you'd like to delete project %s and all of its resources, run %s.`,
  summary.ProjectName,
  o.ProjectName,
  workspace.ProjectDirectoryName,
  color.HighlightCode("ecs-preview project delete"))
return fmt.Errorf("workspace already registered with %s", summary.ProjectName)
```